test/date-time/date-time-create.test.ts

```diff
@@ -0,0 +1,1068 @@
+        when DateTime is created from that codes
+        then  it should throw a validation error`,
+            () =>
+            {
+                checkIsInvalid(validDateCode, "10:00");
+            }
+        );
+
+        test(`Given a valid date code(20240101) and zone, and time code in invalid format 001
+        when DateTime is created from that codes
+        then  it should throw a validation error`,
+            () =>
+            {
+                checkIsInvalid(validDateCode, "001");
+            }
+        );
+
+        test(`Given a valid date code(20240101) and zone, and time code in invalid format 10000
+        when DateTime is created from that codes
+        then  it should throw a validation error`,
+            () =>
+            {
+                checkIsInvalid(validDateCode, "10000");
+            }
+        );
+
+        test(`Given a valid date code(20240101) and zone, and time code with invalid minute 1060
+        when DateTime is created from that codes
+        then  it should throw a validation error`,
+            () =>
+            {
+                checkIsInvalid(validDateCode, "1060");
+            }
+        );
+
+        test(`Given a valid date code(20240101) and zone, and time code with invalid hour 2400
+        when DateTime is created from that codes
+        then  it should throw a validation error`,
+            () =>
+            {
+                checkIsInvalid(validDateCode, "2400");
+            }
+        );
+
+        test(`Given a valid time code(1000) and zone, and date code with invalid day 0 20240100
+        when DateTime is created from that codes
+        then  it should throw a validation error`,
+            () =>
+            {
+                checkIsInvalid("20240100", validTimeCode);
+            }
+        );
+
+        test(`Given a valid time code(1000) and zone, and date code with invalid day 32 20240132
+        when DateTime is created from that codes
+        then  it should throw a validation error`,
+            () =>
+            {
+                checkIsInvalid("20240132", validTimeCode);
+            }
+        );
+
+        test(`Given a valid time code(1000) and zone, and date code with invalid day feb 29 on non-leap year 20230229
+        when DateTime is created from that codes
+        then  it should throw a validation error`,
+            () =>
+            {
+                checkIsInvalid("20230229", validTimeCode);
+            }
+        );
+
+        test(`Given a valid time code(1000) and zone, and date code with invalid day feb 30 20240230
+        when DateTime is created from that codes
+        then  it should throw a validation error`,
+            () =>
+            {
+                checkIsInvalid("20240230", validTimeCode);
+            }
+        );
+
+        test(`Given a valid time code(1000) and zone, and date code feb 29 on leap year 20240229
```

Comment:
I don't think you need to check these. since this is checked in creation. and you have already tested that

Reply:
All you can check here would just be the format of the datecode and timecode